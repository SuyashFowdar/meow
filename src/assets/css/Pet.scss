$all:('', 0, 4000px);
$sm:('-sm', 0, 767px);
$md:('-md', 768px, 991px);
$lg:('-lg', 992px, 4000px);

$breakpoints: ($all, $sm, $lg);

@mixin sm {
  @media screen and (min-width: nth($sm, 2)) and (max-width: nth($sm, 3)) {
    @content;
  }
}

@mixin md {
  @media screen and (min-width: nth($md, 2)) and (max-width: nth($md, 3)) {
    @content;
  }
}

@mixin lg {
  @media screen and (min-width: nth($lg, 2)) and (max-width: nth($lg, 3)) {
    @content;
  }
}

.pet-item {
  height: calc((100vh - 144px) / 2);

  @include lg {
    width: 25%;
  }

  @include md {
    width: 50%;
  }

  @include sm {
    width: 100%;
  }

  .breed {
    padding: 10px;
    color: white;
    background-color: rgba(0, 0, 0, 0.7);
  }
}
