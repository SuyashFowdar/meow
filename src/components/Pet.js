import React from 'react';

const Pet = () => (
  <div>Pet</div>
);

export default Pet;
