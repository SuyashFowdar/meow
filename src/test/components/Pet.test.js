import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { BrowserRouter as Router } from 'react-router-dom';
import Pet from '../../components/Pet';
import reducer from '../../reducers/index';

it('renders correctly', () => {
  const store = createStore(reducer);
  const tree = renderer
    .create((
      <Provider store={store}>
        <Router><Pet pet={{ pet: {}, breed: {} }} /></Router>
      </Provider>
    ))
    .toJSON();
  expect(tree).toMatchSnapshot();
});
