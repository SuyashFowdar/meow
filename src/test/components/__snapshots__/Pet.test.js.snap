// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<a
  className="pet-item bg-cover col main-end"
  href="/pet/undefined"
  onClick={[Function]}
  style={
    Object {
      "backgroundImage": "url(undefined)",
    }
  }
>
  <div
    className="breed"
  />
</a>
`;
